### DISCUSSION TITLE
Synchronous tax webhook is not called

### DISCUSSION THREAD

**Discussion by rrrnld** [saleor/saleor#12241] [Q&A] [answered]
We're running saleor in kubernetes. One part of our deployment is an app that handles taxes via the synchronous tax webhook. It is set up automatically using the following script:

``` python
import os
import sys

from saleor.app.models import App, AppToken
from saleor.permission.enums import get_permissions_from_codenames, CheckoutPermissions
from saleor.webhook.models import Webhook, WebhookEvent
from saleor.webhook.event_types import WebhookEventSyncType

# Delete previously installed apps to make it easier to deal w/ config changes
App.objects.filter(name="Tax App").delete() 

# create tax app and make sure it's allowed to do what it does
app = App.objects.create(name="Tax App")

permissions = get_permissions_from_codenames([CheckoutPermissions.HANDLE_TAXES.codename])
app.permissions.add(permissions.get().id)

# use a pre-defined token for access management
token = AppToken(app=app, name="Auto-generated token")
token.set_auth_token(os.getenv("SALEOR_TAX_APP_TOKEN"))
token.save()

# create webhook handler
webhook = Webhook.objects.create(
  name="Taxes",
  target_url="http://saleor-tax-app:8080/handler",
  app=app
)

# make sure webhook endpoint is called at the right time
event_type = WebhookEventSyncType.CHECKOUT_CALCULATE_TAXES
webhook.events.create(event_type=event_type)

print("App set up successfully")
```

While updating from 3.8 to 3.11 we noticed in tests that the tax app is not called anymore (its log show that the `/handler` endpoint is not accessed at all). The url `http://saleor-tax-app:8080/handler` is accessible from within the saleor api container, which I verified by starting a shell using `python manage.py shell` and doing requests manually using the python `requests` module.

Did anything change between 3.8 and 3.11 that causes these webhooks to be called differently? Is there an error in my setup code that I'm missing?

Thanks in advance for any help

**Comment by timuric** [comment]
Hey @heyarne I would recommend stop using plugins API and use GraphQL API instead, we are very soon will announce plugin APIs deprecated

**Reply by rrrnld** [reply]
I'm not sure I understand - where do I use the plugins API? I'm setting up an app that is listening via Webhooks? Are you suggesting I set up the app via GraphQL? When I set up this app there is no other app that could make an authorized graphql call, it's a bare-bones Saleor installation.

**Reply by timuric** [reply]
@heyarne sorry I got confused your code with plugins, but in any case I would advise using GraphQL talking to Saleor.

**Reply by rrrnld** [reply]
@timuric  What I'm doing is essentially a headless setup for saleor; I'm spinning up different containers and then executing a couple of scripts to configure saleor as I need it, without any clicks (this is because I'm running it in different environments and want the setup to be as similar as possible).

I can't use graphql to do that, can I? I don't have any app that's authorized to make the required requests at that point, the database is empty and there's no secret token I can use that has the permissions needed for further setup.

**Comment by timuric** [comment, marked as answer]
@heyarne in `configuration -> taxes` do you have channels set to `use tax app`?

**Reply by rrrnld** [reply]
That's it! I have had some old channels that have not been updated, and the default is to use a flat tax rate. Thanks!
